src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -334,6 +334,18 @@ IsCitusTableRTE(Node *node)
 }
 
 
+/*
+ * IsLocalPostgresTableRTE gets a node and returns true if the node is a
+ * range table relation entry that points to a local postgres relation.
+ */
+bool
+IsLocalPostgresTableRTE(Node *node)
+{
+	Oid relationId = NodeTryGetRteRelid(node);
```

Comment:
do we need to be concerned about partitioned tables, views, etc.?

it seems this function is strictly identifying relations

Reply:
That's a good point. On the hook, we don't get Views/Mat views, so we don't qualify those as local tables. Now, we do.

Looking to find a solution

Reply:
The current check in NodeTryGetRteRelid on master is:
```

    if (!(rangeTableEntry->rtekind == RTE_RELATION &&
          rangeTableEntry->relkind == RELKIND_RELATION))
    {
        return InvalidOid;
    }
```

which means it would return InvalidOid for a partitioned/foreign/matview local table when it probably shouldn't

Reply:
That is certainly a bug that we are fixing on #4294. This bug leads to various issues with partitioned/foreign tables